tests/test_cap_api.py

```diff
@@ -238,42 +240,102 @@ def test_create_method_when_validate_failed_raises_exception(
             mocked_cap_api.create(json_='file', ana_type='atlas-workflows')
 
 
-def test_update_method_when_no_file_with_data_given(mocked_cap_api):
-    with raises(IOError):
-        mocked_cap_api.update(pid='some_pid')
+@patch('{}.open'.format(builtin_module_name), new_callable=mock_open)
+def test_update_method_when_json_file_not_found(mock_open, cap_api):
+    mock_open.side_effect = IOError
+
+    with raises(MissingJsonFile):
+        cap_api.update(pid='some_pid', json_='file')
 
 
 @patch('{}.open'.format(builtin_module_name),
        new_callable=mock_open,
        read_data='{,}]')
-def test_update_method_when_no_json_in_given_file(mock_open, mocked_cap_api):
-    with raises(ValueError):
-        mocked_cap_api.update(filename='file')
+def test_update_method_when_no_json_in_given_file(mock_open, cap_api):
+    with raises(MissingJsonFile):
+        cap_api.update(pid='some-pid', json_='file')
 
 
-def test_update_method_when_validate_failed_raises_exception(
-    mocked_cap_api, record_data):
-    json_data = json.dumps(record_data)
-    with patch('{}.open'.format(builtin_module_name),
-               new_callable=mock_open,
-               read_data=json_data):
-        mocked_cap_api._make_request.side_effect = [BadStatusCode(), None]
-        with raises(BadStatusCode):
-            mocked_cap_api.update(filename='file', pid='some_pid')
+@responses.activate
+def test_update_method_when_validate_failed_raises_exception(cap_api):
+    url = 'https://analysispreservation-dev.cern.ch/api/deposits/some-pid'
+    mock_json_data = {
+        "title": "Bad Title",
+        "basic_info": {"abstract": "Example Abstract"}
+    }
+    mock_response = {
+        "status": 400,
+        "message": "Validation error. Try again with valid data",
+        "errors": [{"field": [],
+                    "message": "Additional properties are not allowed ('title' was unexpected)"}]
+    }
+    responses.add(responses.PUT, url,
+                  content_type='application/json',
```

Comment:
as mentioned before, you don't really have to specify content_type when you're passing json (it will be set automatically)
same for all your tests below